request.js

```diff
@@ -1293,13 +1297,25 @@ Request.prototype.onRequestResponse = function (response) {
       self.pipeDest(dest)
     })
 
+    if (typeof self.maxResponseSize === 'number') {
+      var responseBytesLeft = self.maxResponseSize
+    }
+
     responseContent.on('data', function (chunk) {
       if (self.timing && !self.responseStarted) {
         self.responseStartTime = (new Date()).getTime()
 
         // NOTE: responseStartTime is deprecated in favor of .timings
         response.responseStartTime = self.responseStartTime
       }
+      if (typeof responseBytesLeft !== 'undefined') {
+        responseBytesLeft -= chunk.length
+        if (responseBytesLeft < 0) {
+          self.emit('error', new Error('Maximum response size reached'))
```

Comment:
Leave a lot of comment here to not change this message without updating in the app.